messaging/src/main/java/org/axonframework/commandhandling/gateway/AbstractRetryScheduler.java

```diff
@@ -45,16 +45,19 @@
     private static final int DEFAULT_MAX_RETRIES = 1;
 
     private final ScheduledExecutorService retryExecutor;
+    private final Predicate<Throwable> nonTransientFailurePredicate;
```

Comment:
Curious: Why aren't you using the `NonTransientClassesPredicate` type for the `nonTransientFailurePredicate`?